pyanaconda/modules/timezone/installation.py

```diff
@@ -0,0 +1,165 @@
+
+    def _write_etc_adjtime(self):
+        """Write /etc/adjtime contents.
+
+        :raise: TimezoneConfigurationError
+        """
+        if arch.is_s390():
+            # there is no HW clock on s390(x)
+            return
+
+        try:
+            with open(os.path.normpath(self._sysroot + "/etc/adjtime"), "r") as fobj:
+                lines = fobj.readlines()
+        except IOError:
+            lines = ["0.0 0 0.0\n", "0\n"]
+
+        try:
+            with open(os.path.normpath(self._sysroot + "/etc/adjtime"), "w") as fobj:
+                fobj.write(lines[0])
+                fobj.write(lines[1])
+                if self._is_utc:
+                    fobj.write("UTC\n")
+                else:
+                    fobj.write("LOCAL\n")
+        except IOError as ioerr:
+            msg = "Error while writing /etc/adjtime file: %s" % ioerr.strerror
+            raise TimezoneConfigurationError(msg)
+
+
+class ConfigureNTPTask(Task):
+    """Installation task for NTP configuration."""
+
+    def __init__(self, sysroot, ntp_enabled, ntp_servers):
+        """Create a new task.
+
+        :param str sysroot: a path to the root of the installed system
+        :param bool ntp_enabled: is NTP enabled or not
+        :param ntp_servers: list of NTP servers and pools
+        :type ntp_servers: list of str
+        """
+        super().__init__()
+        self._sysroot = sysroot
+        self._ntp_enabled = ntp_enabled
+        self._ntp_servers = ntp_servers
+
+    @property
+    def name(self):
+        return "Configure NTP"
+
+    def run(self):
+        """Perform the actual work of setting up NTP."""
+        if not (self._ntp_enabled and self._ntp_servers):
+            return
+
+        chronyd_conf_path = os.path.normpath(self._sysroot + ntp.NTP_CONFIG_FILE)
+        pools, servers = ntp.internal_to_pools_and_servers(self._ntp_servers)
+
+        if os.path.exists(chronyd_conf_path):
+            log.debug("Modifying installed chrony configuration")
+            try:
+                ntp.save_servers_to_config(pools, servers, conf_file_path=chronyd_conf_path)
+            except ntp.NTPconfigError as ntperr:
+                log.warning("Failed to save NTP configuration: %s", ntperr)
+
+        # use chrony conf file from installation environment when
+        # chrony is not installed (chrony conf file is missing)
+        else:
+            log.debug("Creating chrony configuration based on the "
+                      "configuration from installation environment")
+            try:
+                ntp.save_servers_to_config(pools, servers,
+                                           conf_file_path=ntp.NTP_CONFIG_FILE,
+                                           out_file_path=chronyd_conf_path)
+            except ntp.NTPconfigError as ntperr:
+                log.warning("Failed to save NTP configuration without chrony package: %s",
+                            ntperr)
```

Comment:
What I write here is only food for thought. I don't expect you would change this PR immediately.

The structure here is basically:
```
if:
... something with try-except ....

# comment what is this else clause doing
else:
.... something else with try-except ....
```

When I'm looking on this I'm thinking why is the comment there? If the code have to be explained then why do not create a method  for that with a self-describing name.

```
if:
self._configure_installed_chrony()
else:
self._configure_when_chrony_is_not_installed()
```

Reply:
Hm, that would add more boilerplate than the removed comments, though: 7>2...

Reply:
It's not about lines counting but about readability.